octodns/manager.py

```diff
@@ -244,6 +227,37 @@ def __init__(self, config_file, max_workers=None, include_meta=False):
                     'Incorrect plan_output config for ' + plan_output_name
                 )
 
+        self._zone_tree = None
+
+    @property
+    def zone_tree(self):
+        if self._zone_tree is None:
+            zone_tree = {}
+
+            # Get a list of all of our zone names
+            zones = list(self.config['zones'].keys())
+            # Sort them from shortest to longest so that parents will always
+            # come before their subzones
+            zones.sort(key=lambda z: len(z))
+            # Until we're done processing zones
+            while zones:
+                # Grab the one we'lre going to work on now
+                zone = zones.pop(0)
+                trimmer = len(zone) + 1
+                subs = set()
+                # look at all the zone names that come after it
+                for candidate in zones:
+                    # If they end with this zone's name them they're a sub
+                    if candidate.endswith(zone):
+                        # We want subs to exclude the zone portion
+                        subs.add(candidate[:-trimmer])
+
+                zone_tree[zone] = subs
+
+            self._zone_tree = zone_tree
+
+        return self._zone_tree
```

Comment:
The output of this is a bit harder to parse in my opinion as it is no-longer tree-like.
Certainly easier to grab children of a zone however as it's a simple map however.

Previously
```
{
  'unit.tests': {
    'skipped.alevel': {},
    'sub': {
        'another': {}
    }
  }
}
```

Replacment:
```
{
  'unit.tests.': {
    'skipped.alevel',
    'sub', 
    'another.sub'  # < Should this one be listed here, as it's a subzone of `sub`
  }, 
  'sub.unit.tests.': {
    'another'
  },
  'another.sub.unit.tests.': set(), 
  'skipped.alevel.unit.tests.': set()
}
```

Reply:
Yeah. Looking back at it now I didn't like that the computation/logic happened in two different places/times. That made changes in one place break the other etc which is 😢.

Good point that `zone_tree` no longer makes sense as the name. I'll clean that up. Looks like zone_tree is only used during `configured_sub_zones` so the right solution is probably to ditch the property and do the computing in `configured_sub_zones` itself.  It think that'll even let things do it per zone and only when needed.